jstz_node/src/services/logs/broadcaster.rs

```diff
@@ -0,0 +1,91 @@
+use std::{sync::Arc, time::Duration};
+
+use actix_web::rt::time::interval;
+use actix_web_lab::{
+    sse::{self, Sse},
+    util::InfallibleStream,
+};
+use futures_util::future;
+use parking_lot::Mutex;
+use tokio::sync::mpsc;
+use tokio_stream::wrappers::ReceiverStream;
+
+pub struct Broadcaster {
+    clients: Mutex<Vec<mpsc::Sender<sse::Event>>>, // TODO: Use a read-write lock instead?
+}
```

Comment:
Is it possible to put the `Arc` inside this struct? Like:
```suggestion
pub struct Broadcaster {
    clients: Arc<Mutex<Vec<mpsc::Sender<sse::Event>>>>,
}
```

Reply:
Additionally, `Broadcaster` could then implement a cheap `Clone`

Reply:
I've tried but in that case we will end up having two `Arc` since `Data` is esenttialy an `Arc`
```
pub struct Broadcaster {
    clients: Arc<Mutex<Vec<mpsc::Sender<sse::Event>>>>,
}

impl<T> Data<T> {
    /// Create new `Data` instance.
    pub fn new(state: T) -> Data<T> {
        Data(Arc::new(state))
    }
}
```
so
```
Data::new(Broadcaster)
```
is going to be 
```
Arc<Boradcaster {cllients:Arc::new(...)}>
```